### DISCUSSION TITLE
TwoPaneView confusion

### DISCUSSION THREAD

**Discussion by varyamereon** [dotnet/maui#16648] [Q&A]
I am trying to implement TwoPaneView for the first time in a new app and I'm not sure if it just isn't working or if I'm doing something wrong. Depending on different devices, the `Mode` property returned is often incorrect, leading to the UI not being displayed correctly.

I am wondering home the mode is calculated on a single screen device, for example if I set the `MinTallModeHeight` to 1000 and the `MinWideModeWidth` to 1000, and the device has a display of 1200 x 2000, how is it determined whether Wide mode or Tall mode should be assigned, as both would be applicable?

I'm not sure whether or not this is the cause of the issues I'm having but thought it worth asking.

**Comment by tkouba** [comment]
I use derived class from `TwoPaneView` with overriden `OnSizeAlloocated`, where there is detected portrait/landscape size of `TwoPaneView` and force update `MinWideModeWidth` and `MinTallModeHeight`. 

```csharp
protected override void OnSizeAllocated(double width, double height)
{
    if (width > height * 1.1) 
    {
        // Landscape
        MinWideModeWidth = 1;
        MinTallModeHeight = 2 * height;
    }
    else
    {
        // Portrait
        MinTallModeHeight = 1;
        MinWideModeWidth = 2 * width;
    }
    base.OnSizeAllocated(width, height);
}
```
